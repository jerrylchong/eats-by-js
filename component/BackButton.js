import React from 'react';
import {Image, TouchableOpacity, StyleSheet, View, Dimensions} from "react-native";

const BackButton = (props) => {
    return (
        <View style = {props.style}>
            <TouchableOpacity style = {styles.container} onPress = {props.onPress}>
                {props.white
                    ? <Image style = {styles.image} source ={require('../assets/whiteback.png')} />
                    : <Image style = {styles.image} source ={require('../assets/backbutton.png')} />
                }
            </TouchableOpacity>
        </View>
    )
}

export default BackButton

const windowWidth= Dimensions.get('window').width;

const styles = StyleSheet.create({
    container: {
        height: windowWidth * 0.1,
        width: windowWidth * 0.1,
        alignItems: 'flex-start',
        justifyContent: 'flex-start'
    },
    image: {
        height: 13,
        width: 15
    }
})