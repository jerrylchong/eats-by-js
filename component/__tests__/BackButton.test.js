import React from 'react';
import { TouchableOpacity, View, Image } from 'react-native';
import BackButton from "../BackButton";

import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(<BackButton/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });

test('test component rendering', () => {
    const wrapper = shallow(<BackButton/>);
    expect(wrapper.find(View)).toBeTruthy(); // checks if view is rendered
    expect(wrapper.find(TouchableOpacity)).toBeTruthy(); // checks if touchable opacity is rendered
    expect(wrapper.find(Image)).toBeTruthy(); // checks if image is rendered
})

test('test onPress functionality', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked'); // create mock function
    const wrapper = shallow(<BackButton onPress={onPressEvent}/>);
    wrapper.find(TouchableOpacity).props().onPress(); // press TouchableOpacity
    expect(onPressEvent.mock.calls.length).toBe(1); // expect 1 mock function call
})