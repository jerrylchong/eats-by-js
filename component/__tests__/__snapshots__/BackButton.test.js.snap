// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "height": 50,
        "justifyContent": "flex-start",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 13,
          "width": 15,
        }
      }
    />
  </View>
</View>
`;
